src/common/lib/client/baserealtime.ts

```diff
@@ -142,6 +143,33 @@ class ConsumerGroup extends EventEmitter {
     }
   }
 
+  async leave() {
+    if (!this.consumerGroupName) {
+      return;
+    }
+    try {
+      Logger.logAction(
+        Logger.LOG_MAJOR,
+        'ConsumerGroup.leave()',
+        'leaving consumer group ' + this.consumerGroupName + ' as ' + this.consumerId
+      );
+      if (!this.channel) {
+        Logger.logAction(Logger.LOG_ERROR, 'ConsumerGroup.leave()', 'leave called with no channel initialised');
+        return;
+      }
+      this.channel.presence.unsubscribe(this.computeMembershipListener);
+      await this.channel.presence.leave(null);
+      await this.channel.detach();
+      this.channel = undefined;
+    } catch (err) {
+      Logger.logAction(
+        Logger.LOG_ERROR,
+        'ConsumerGroup.leave()',
+        'failed to leave presence set on consumer group channel:' + err
```

Comment:
I think you usually do `Utils.inspectError(err)` if you want to append err to log message.
Might want to change other places too in another PR (saw it in ChannelGroup.addSubscriptions too).